axelrod/strategies/handshake.py

```diff
@@ -0,0 +1,33 @@
+from axelrod import Actions, Player, Game
+
+C, D = Actions.C, Actions.D
+
+
+class Handshake(Player):
+    """Starts with C, D. If the opponent plays the same way, cooperate forever,
+    else defect forever."""
+
+    name = 'Handshake'
+    classifier = {
+        'memory_depth': float('inf'),  # Long memory
+        'stochastic': False,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        Player.__init__(self)
+        self.initial_plays = [C, D]
```

Comment:
There's probably merit in opening an issue to take stock of all the
strategies and see what can be parametrized.

I'd prefer to see this done here already (seems like it might as well) but
up to you/second reviewer.

Hoping to get to this properly this evening :)

On Sun, 15 May 2016, 20:25 Marc Harper, <notifications@github.com> wrote:

> In axelrod/strategies/handshake.py
> <https://github.com/Axelrod-Python/Axelrod/pull/591#discussion_r63296633>:
>
> > +    """Starts with C, D. If the opponent plays the same way, cooperate forever,
> > +    else defect forever."""
> > +
> > +    name = 'Handshake'
> > +    classifier = {
> > +        'memory_depth': float('inf'),  # Long memory
> > +        'stochastic': False,
> > +        'makes_use_of': set(),
> > +        'inspects_source': False,
> > +        'manipulates_source': False,
> > +        'manipulates_state': False
> > +    }
> > +
> > +    def __init__(self):
> > +        Player.__init__(self)
> > +        self.initial_plays = [C, D]
>
> There's no drawback, I just added the strategy for completeness with the
> literature. We can generalize now if you'd like, or leave it as an easy PR
> for someone else.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/Axelrod-Python/Axelrod/pull/591/files/168ad6a6d0976312da1c80719d4c6980740d38b8#r63296633>
>